tests/integration/vlan_test.py

```diff
@@ -62,6 +65,53 @@ def test_set_vlan_iface_down(eth1_up):
         assert not current_state[INTERFACES]
 
 
+def test_add_down_remove_vlan(eth1_up):
+    with yaml_state('vlan101_eth1_up.yml',
+                    cleanup='vlan101_eth1_absent.yml') as desired_state:
+        assertlib.assert_state(desired_state)
+        with yaml_state('vlan101_eth1_down.yml') as desired_state:
+            assertlib.assert_absent(VLAN_IFNAME)
+
+    assertlib.assert_absent(VLAN_IFNAME)
+
+
+@contextmanager
+def yaml_state(initial, cleanup=None):
```

Comment:
I would prefer only the `load_example` to know about the input format (yaml), all the rest can be used with regular dicts.

Reply:
We can split out a generic method when we need it but I would like to also have a yaml one to avoid having to load all the examples explicitly in the tests.

Reply:
IMO the logic of the contextmanager should not be mixed with the format the the config resides in.
You can use fixtures for the config reading if you do not want to explicitly read the config from within the test. But I see no problem in explicitly telling in the test from where you get the config/state.

Reply:
Not sure if we are discussing the same thing.
In the tests I am fine with specifying the file name of the state files but I am not fine with loading the files there as well, since this is something that the context manager can do for me.

About mixing the format: If the context manager is for dictionaries, then the logic also mixes the requirement of getting a dictionary instead of getting a filename with the remaining logic.

If you need this context manager also with pure dictionary support, we can add it and build one for yaml on top of it. But I would prefer to do this in a seperate step when we need it.